src/magicgui/widgets/_table.py

```diff
@@ -269,6 +269,39 @@ def value(self, value: TableData) -> None:
         for row, d in enumerate(data):
             self._set_rowi(row, d)
 
+    def delete_row(
+        self,
+        *,
+        index: int | Sequence[int] | None = None,
+        header: Any | Sequence[Any] | None = None,
+    ) -> None:
+        """Delete row(s) by index or header.
+
+        Parameters
+        ----------
+        index : int or Sequence[int], optional
+            Index or indices of row(s) to delete.
+        header : Any or Sequence[Any], optional
+            Header or headers of row(s) to delete.
+        """
+        indices: set[int] = set()
+        if index is not None:
+            if isinstance(index, Sequence):
+                indices.update(index)
+            else:
+                indices.add(index)
+        if header is not None:
+            if isinstance(header, str) or not isinstance(header, Sequence):
```

Comment:
Oh man, I needed a double-take to understand why this wasn't just `not isinstance(header, Sequence)`. That's annoying 😂

Reply:
Yep, classic :) string is a sequence but not "that" kind :)